inginious/frontend/plugins/show_hints/static/js/hints_edit.js

```diff
@@ -0,0 +1,176 @@
+var update_hint_id = null;
+
+/* Show a message into an alert */
+function displayAlert(alert_container_id, message, duration){
+    alert = $("#"+alert_container_id);
+    alert.text(message);
+    alert.show();
+    setTimeout(function(){
+        clearAlert(alert_container_id);
+    }, duration);
+}
+
+/* Clear and hide modal alert*/
+function clearAlert(alert_id){
+    let alert = $("#"+alert_id);
+    alert.fadeTo(100, 1).slideUp(250, () => {
+        alert.slideUp(250);
+        alert.text("");
+    });
+}
+
+/* Load code editor for the hint content on modal */
+function loadHintContentCodeEditor(){
+    $("#hint_content_container").show();
+
+    let hints_editor = codeEditors["task_hints_content"];
+    hints_editor.refresh();
+}
+
+function onSaveHintClick(){
+
+    let hint = $.extend({
+        "title" : "",
+        "penalty" : 0.0,
+        "content" : ""
+    },{
+        "title" : $("#hint_title")[0].value,
+        "penalty" : parseFloat($("#hint_penalty")[0].value),
+        "content" : $("#hint_content")[0].value
+    });
+
+    /* Check if the entries are allowed on the hint */
+
+    let contentBlankEntry = hint["content"].match(/^\s*$/)
+
+    if(!hint["title"] || contentBlankEntry){
+        const message = "You need to complete mandatory fields";
+        let editAlert = "hint_edit_alert";
+        displayAlert(editAlert, message, 10000);
+        return ;
+    }
+    if(isNaN(hint["penalty"])){
+        const message = "Hint penalty needs to be a number";
+        let editAlert = "hint_edit_alert";
+        displayAlert(editAlert, message, 10000);
+        return ;
+    } else if (0 > hint["penalty"] || hint["penalty"] > 100){
+        const message = "Hint penalty must be between 0.0% and 100.0%";
+        let editAlert = "hint_edit_alert";
+        displayAlert(editAlert, message, 10000);
+        return ;
+    }
```

Comment:
This checks must also be done in the python code where you manage the request. Image that someone might not use the UI, and use something like Postman to send requests, thus these kind of checks must be done in the backend too.